ably/realtime_channel.go

```diff
@@ -114,6 +125,46 @@ func (ch *RealtimeChannels) Get(name string, options ...ChannelOption) *Realtime
 	return c
 }
 
+// GetDerived creates a new derived [ably.RealtimeChannel] object for given channel name, using the provided derive options and
+// channel options if any. Returns error if any occurs
+func (ch *RealtimeChannels) GetDerived(name string, deriveOptions DeriveOptions, options ...ChannelOption) (*RealtimeChannel, error) {
+	if deriveOptions.Filter != "" {
+		match, err := matchDerivedChannel(name)
+		if err != nil {
+			return nil, err
+		}
+		filter := url.PathEscape(deriveOptions.Filter)
+		name = fmt.Sprintf("[filter=%s%s]%s", filter, match.qualifierParam, match.channelName)
+	}
+	return ch.Get(name, options...), nil
+}
+
+// This regex check is to retain existing channel params if any e.g [?rewind=1]foo to
+// [filter=xyz?rewind=1]foo. This is to keep channel compatibility around use of
+// channel params that work with derived channels.
+func matchDerivedChannel(name string) (*derivedChannelMatch, error) {
```

Comment:
Is it worth adding a unit test for this function to verify all the possible cases are handled? (qualifiers, channel params etc)